qa/tasks/workunit.py

```diff
@@ -16,6 +16,55 @@
 log = logging.getLogger(__name__)
 
 
+class Refspec:
+    def __init__(self, refspec):
+        self.refspec = refspec
+
+    def __str__(self):
+        return self.refspec
+
+    def _clone(self, git_url, clonedir, opts=[]):
+        return (['rm', '-rf', clonedir] +
+                [run.Raw('&&')] +
+                ['git', 'clone'] + opts +
+                [git_url, clonedir])
+
+    def _cd(self, clonedir):
+        return ['cd', '--', clonedir]
```

Comment:
?  I don't see -- in cd's arguments, and surely we can count on clonedir not starting with '-'?